src/EditorFeatures/Core/Implementation/Peek/DefinitionPeekableItem.cs

```diff
@@ -72,7 +72,7 @@ public void FindResults(string relationshipName, IPeekResultCollection resultCol
                 if (!sourceLocations.Any())
                 {
                     // It's a symbol from metadata, so we want to go produce it from metadata
-                    var declarationFile = _peekableItem._metadataAsSourceFileService.GetGeneratedFileAsync(project, symbol, cancellationToken).WaitAndGetResult(cancellationToken);
+                    var declarationFile = _peekableItem._metadataAsSourceFileService.GetGeneratedFileAsync(project, symbol, allowDecompilation: false, cancellationToken).WaitAndGetResult(cancellationToken);
```

Comment:
Why false here? The expectation here is both behave the same. That *might* that this dialog should be pushed into the metadata as source file service, or maybe that gets a separate function GetGeneratedFileThatAsksIfYouWantToDecompileOrNotAsync()?

Reply:
➡️ Currently I haven't incorporated the legal notice at this entry point. Also the legal notice does not remember a "Yes" selection.

❓ Is there a way to make an `Option<bool>` that defaults to false and is not persisted? If so, I could make a "Yes" selection for the legal notice persist for the session at least.

Reply:
If you say "yes", do we say yes for all future assemblies in the session, or just that assembly? I think we discussed it but I forget the result. If anything, the legal notice hopefully can make it clearer what you're "agreeing" to.

Reply:
➡️ Filed #23479 for the legal notice option

➡️ Filed #23480 for Peek Definition

Reply:
If you don't want it to persist, just don't specify a StorageLocation.

Reply:
Ick.  user needs to approve hte notice once per session?  Is there a reason to not have a "i understand, never tell me again" option?